datastore/longhorn.go

```diff
@@ -117,17 +121,24 @@ func (s *DataStore) updateSettingDefinitions(customizedDefaultSettings map[strin
 }
 
 func (s *DataStore) createOrUpdateSetting(name types.SettingName, definition types.SettingDefinition) error {
+	// Skip the create or update to avoid the validation error from ValidateSetting(),
+	// if the required setting is not updated somewhere.
+	if definition.Required && definition.Current == "" {
+		return nil
+	}
```

Comment:
Should it just be?

```
	if definition.Required{
		return nil
	}
```

Reply:
This function is to create or update settings according the values collected from setting CRs or configmap.
In general, the required field should be created and updated with a given non-empty value.
If it is an empty value, we skip them here. Thus, we need to check `definition.Current` here.

Reply:
I think I misread the code. Thanks for the explanation.